pkg/apis/pipeline/v1alpha1/git_resource.go

```diff
@@ -30,6 +34,25 @@ type GitResource struct {
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 }
 
+// NewGitResource create a new git resource to pass to Knativve Build
+func NewGitResource(r *PipelineResource) *GitResource {
+	gitResource := GitResource{
+		Name: r.Name,
+		Type: r.Spec.Type,
+	}
+	for _, param := range r.Spec.Params {
+		switch {
+		case strings.EqualFold(param.Name, "URL"):
+			gitResource.URL = param.Value
+		case strings.EqualFold(param.Name, "serviceAccount"):
+			gitResource.ServiceAccount = param.Value
+		case strings.EqualFold(param.Name, "Revision"):
+			gitResource.Revision = param.Value
+		}
+	}
```

Comment:
Also, it should check if url is defined  and is correct or this is handled in PipelineResource validation?

Reply:
I was expecting this validation to be done in the webhook before the object is created